packages/@ama-sdk/core/src/fwk/mocks/helpers.ts

```diff
@@ -22,20 +22,22 @@ export function getOperationId(pathObject: PathObject, method: string): string |
  *
  * @param requestUrl the URL string
  * @param pathObjects the list of available path objects
+ * @param method the optional HTTP method used in case of several matches
  */
-export function getPath(requestUrl: string, pathObjects: PathObject[]): PathObject | undefined {
+export function getPath(requestUrl: string, pathObjects: PathObject[], method?: string): PathObject | undefined {
   const pathName = new URL(requestUrl, requestUrl.startsWith('/') ? 'http://example.com' : undefined).pathname;
 
   if (!pathName) {
     throw new Error(`Couldn't parse url ${requestUrl}`);
   }
 
   // Find all matching paths
-  let matches = pathObjects.reduce<{index: number; segments: string[]}[]>((newMatches, pathObject, index) => {
+  let matches = pathObjects.reduce<{index: number; segments: string[]; methods: string[]}[]>((newMatches, pathObject, index) => {
```

Comment:
Instead of `string` you can use literal union type to avoid the following `toLowerCase`

Reply:
The initial idea was to restrict the method type to HTTP methods `get`, `post`, etc, But as discussed with @kpanot, each objects in this context are typed with a `string` type (both `PatchObject.operation.method` and `EncodedApiRequest.method` are of type `string`) => Too much changes for this contribution. The `toLowerCase` will be kept, as it is already used by others methods of the helper.